protocol/op-deployer-embedded-artifacts.md

```diff
@@ -0,0 +1,67 @@
+reasons:
+
+- The artifacts that a tagged locator points to are created from a different branch than OP Deployer itself. This
+  can lead to a mismatch between OP Deployer's deployment code and the artifacts it is trying to deploy.
+- There is a 1:1 mapping between each OP Deployer version and the tagged versions it supports. `v0.2.0` of OP
+  Deployer, for example, only supports smart contracts from `op-contracts/v2.0.0`. However, the version of the smart
+  contracts can sometimes change by minor or patch versions. This makes it difficult to determine which version of
+  the contracts each version of OP Deployer supports, and can lead to unexpected behavior or broken deployments.
+
+In addition, maintaining the tagged locators themselves is non-trivial amount of work:
+
+- New releases require contract artifacts to be built, pushed to GCP, and added to `standard.go`. This means that
+  every new contract release requires a corresponding code change to OP Deployer.
+- Because the artifacts each locator points to are on GCP, they need to be validated client-side before being used.
+  This adds complexity, overhead, and potential for security vulnerabilities.
+- It's much less clear which commit a given set of artifacts belong to. Artifacts are built using a
+  content-addressable hash of their source code. A commit that doesn't change the source code won't change the hash,
+  so the commit for a given contract release may not match the commit in the uploaded artifacts. While not a huge
+  issue, this makes determining the provenance of a given set of artifacts quite difficult.
+
+## Proposed Solution
+
+Rather than continuing to support tagged locators, we propose embedding the contract artifacts for a given release
+directly inside the OP Deployer binary itself using `go:embed`. HTTPs and file locators will continue to work as
+before. The CLI will select the tagged locator by default if it exists, so users will not generally have to specify
+a locator at all when invoking OP Deployer.
+
+This change would increase the size of the OP Deployer binary by ~60MB after removing test code from the artifacts.
+It is our belief that this is an acceptable tradeoff, particularly since users would have to download the artifacts
+anyway.
+
+For local development, we can make the embedding optional. This will avoid increasing Go compile times when we're not
+cutting an official OP Deployer release.
+
+As part of this solution, we will start developing OP Deployer on the same proposal branches as the contracts 
+themselves. This way, OP Deployer is released side-by-side with the contracts, and we can ensure that they always 
+stay in sync.
+
+## Impact on Developer Experience
+
+Tools like Kurtosis will continue to be able to use HTTPs and file locators. If users want to use a tagged locator,
+they will need to change the version of OP Deployer they use rather than change the locator itself. This will likely
+impact Netchef the most.
```

Comment:
Which kinds of contract changes in https/file locators compared to the tagged contract releases do we expect to be supported by the same op-deployer version? If an https/file locator is used, is it expected that a non-tagged version of `op-deployer` will often be used?